File: java/unit.test.playground/src/main/java/edward/wilde/unit/test/playground/model/Employee.java
package edward.wilde.unit.test.playground.model;

import edward.wilde.unit.test.playground.services.*;
public class Employee {
	private IService<Employee> service;
	
	public Employee(IService<Employee> service) {
		this.service = service;
	}
	
	public void Save() {
		this.service.Save(this);
	}
}

File: c#/domain/Model/Employee.cs
// -----------------------------------------------------------------------
// <copyright file="Employee.cs">
// Copyright (c) 2013.
// </copyright>
// -----------------------------------------------------------------------
namespace domain.Model
{
    using domain.Services;

    public class Employee
    {
        private readonly IService<Employee> service;

        public Employee(IService<Employee> service)
        {
            this.service = service;
        }

        public void Save()
        {
            this.service.Save(this);
        }
    }
}